torch/export/__init__.py

```diff
@@ -597,11 +595,34 @@ class Constraint(_ConstraintTarget):
 
     """
 
-    # NOTE(avik): In the future, this could be Union[StrictMinMaxConstraint, <other kinds>]
     constraint_range: StrictMinMaxConstraint
-    # Represent that `constraint_range` is shared with another _ConstraintTarget, which
-    # typically arises because of a specified equality with another dynamic dimension.
-    shared: Optional[_ConstraintTarget] = None
+    shared: Optional[_ConstraintTarget]
+
+    def __init__(self):
```

Comment:
A modern way of doing this is as follows: https://replit.com/@AvikChaudhuri/MustyDistantComputer#main.py

Basically you use a metaclass and override its `__call__` method, and then you don't have to mess with the __init__ at all.